valhalla/valhalla.h

```diff
@@ -3,4 +3,3 @@
 #define VALHALLA_VERSION_MAJOR 3
```

Comment:
once upon a time i had a bunch of scripting to do autotools style debian packaging for valhalla. it was great, we had debian packages on ppa. then we migrated to cmake and all that stuff didnt work but we kept the template headers

at this point you are right we dont need them anymore. actually i think we should drop `config.h` completely lets just move the packaging stuff into `valhalla.h` as well and simplify all the way. what do you think @nilsnolde ?

Reply:
ah wait i guess we cant move PACKAGE and PACKAGE_VERSION they need to stay in config.h as they would collide in other projects. we should also check that cmake is getting these variables properly for its packaging targets

Reply:
the version variables make it into the main cmakelists.txt